contracts/multisig-prover/src/payload.rs

```diff
@@ -70,6 +80,23 @@ impl Payload {
                 abi::execute_data::encode(verifier_set, signers_with_sigs, &payload_hash, payload)
             }
             Encoder::Bcs => todo!(),
+            Encoder::Solana => {
+                let mut enc_signatures = Vec::new();
+
+                for s in signers_with_sigs {
+                    enc_signatures.push(to_weighted_signature(&s)?)
+                }
+
+                let bytes = axelar_encoding::encode::<1024>(
```

Comment:
An idea on how to reason with the magic number:
1. we need to cap the amount of of signers in the signer set
2. we need to cap the amount of messages in the payload (this might not work if the "payload" contains the actual payload bytes rather than just the hash of the payload)
3. use the "worst case scenario" of max signers and max message amount to determine the size

I remember @tilacog mentioning that changing the number to smaller ones (e.g `0`) has no real effect. Do you mind verifying this?

Reply:
I think your idea makes completely sense, please check this research issue and add your ideas there: https://github.com/eigerco/solana-axelar-internal/issues/332